Automattic/jetpack
# Backup: provide additional disconnect info in backup plugin

This PR is related to and based from #21048

#### Changes proposed in this Pull Request:

* Include data about other plugins using the Jetpack connection in the initial state for the Jetpack Backups app.
* Include analytics scripts for the `<DisconnectDialog />`.

#### Jetpack product discussion
pbNhbs-V4-p2

#### Does this pull request change what data or activity we track or use?
This PR does not track any new data or activity on its own.

#### Testing instructions:
* Review PR #21048 to get an overview of the updates to the disconnect dialog component that this PR uses.
* On your development Jetpack environment, make sure you have the Jetpack Backup plugin installed and activated.
* Make sure you also have at least one other plugin that uses the Jetpack connection installed and active ( Boost, Jetpack )
* Go to the Jetpack Backup admin screen.
* If not already connected, follow the connection flow to connect the Jetpack site.
* Once connected, from the Jetpack Backup admin page, click the "Disconnect" link in the connection area.
* This will open the Disconnect Dialog.
* You should see a list of other connected plugins presented in the first screen that looks like the example below:
![Screen Shot 2021-10-14 at 12 13 42 PM](https://user-images.githubusercontent.com/18016357/137358560-4a00c0cc-8b01-4433-8802-b9da3f7eb628.png)

* After a successful disconnect, you will see the disconnect confirmation screen, which will provide the option to give some feedback. You can also exit the flow here by clicking on "No thank you"
![Screen Shot 2021-10-22 at 3 55 35 PM](https://user-images.githubusercontent.com/18016357/138524914-c9f4c238-814e-4bd3-8507-d1bb5a50da1d.png)

* Click on the "Help us improve" button to go to the survey screen.
![Screen Shot 2021-10-14 at 12 05 25 PM](https://user-images.githubusercontent.com/18016357/137379571-840434ea-daaa-48fd-b407-cbfb0b42e701.png)

* On the survey screen, the "Submit Feedback" button should be disabled until a response is selected. Selecting a response should show the response in an active/ selected state that looks like this:
![Screen Shot 2021-10-14 at 12 05 37 PM](https://user-images.githubusercontent.com/18016357/137379602-977d30e0-509b-49e5-90eb-0cd1adff87e9.png)

* Press "Submit Feedback" to submit the response and move to the last step. ( note, the disconnect flow can be exited here without providing feedback by clicking on "Skip for now" )
* The survey response is sent to a WPCOM endpoint. Observe your network panel traffic for a POST request to a `feedback-survey` endpoint. You should receive a 200 response from this endpoint.

* The final step of the flow is the "Thank You" screen. You can click "Back to my website," which will exit the disconnect flow and call the passed `onDisconnect` prop - if provided.
![Screen Shot 2021-11-20 at 11 25 49 AM](https://user-images.githubusercontent.com/18016357/142734288-44ebe9d2-cb27-4944-af2d-3b267f8f687b.png)

* Check that the analytics events from the test plan for #21048 are recorded by tracks.
* After exiting the disconnect flow, you should see the Connect screen for the backups plugin:
![Screen Shot 2021-10-22 at 5 29 50 PM](https://user-images.githubusercontent.com/18016357/138525104-b6cbd911-4afb-47a7-b5b7-1d64bcdff775.png)
